### PR TITLE
Basic project setup and Travis

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Closes https://github.com/giantswarm/formica/issues/5

### CODE REVIEW

**Comment by JosephSalisbury** [comment]
@zeisss @xh3b4sd what do you guys think?

On Travis, we build in their container (with go 1.5 specified), and locally we build within the golang:1.5 image. Best of both worlds! :D

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+$(BIN): $(SOURCE) VERSION .gobuild
+	echo Building inside Docker container for $(GOOS)/$(GOARCH)
+	docker run \
+	    --rm \
+	    -v $(shell pwd):/usr/code \
+	    -e GOPATH=/usr/code/.gobuild \
+	    -e GOOS=$(GOOS) \
+	    -e GOARCH=$(GOARCH) \
+	    -w /usr/code \
+	    golang:1.5 \
```
Comment (xh3b4sd) [inline Makefile:44]: @hectorj2f had a comment about the different go versions we use. We should align these.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+BUILD_PATH := $(shell pwd)/.gobuild
+GS_PATH := $(BUILD_PATH)/src/github.com/giantswarm
+GOPATH := $(BUILD_PATH)
+
+BIN := $(PROJECT)
+
+VERSION := $(shell cat VERSION)
+COMMIT := $(shell git rev-parse --short HEAD)
+
+.PHONY: clean
```
Comment (xh3b4sd) [inline Makefile]: There are some targets missing here.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+language: go
+
+go:
+    - 1.5
+
+before_install:
+- wget https://downloads.giantswarm.io/builder/0.11.0/builder
+- chmod +x ./builder
+- export PATH=$PATH:$PWD
```
Comment (xh3b4sd) [inline .travis.yml:11]: Builder is currently not used. Do you want to use it later for pinning versions?

Comment (JosephSalisbury) [inline .travis.yml:11]: Yeah, we'll need it as soon as we start using external packages.

File: Makefile (hunk 1 above)
Comment (JosephSalisbury) [inline Makefile:44]: I've gone for go 1.5 currently, and am open to change - I think this is a larger discussion though.

**Comment by zeisss** [comment]
I am fine with all of this.

**Pull request merged**